setup.py

```diff
@@ -0,0 +1,48 @@
+from setuptools import setup, Distribution
+import glob
+import os
+import platform
+import sys
+
+installDir = ''
+for child in os.listdir(os.getcwd()):
+    if os.path.isdir(child):
+        subdirs = os.listdir(child)
+        if 'tests' in subdirs and 'bin' in subdirs:
+            installDir = child
+
+packages = ''
+if platform.system() == 'Windows':
+    packages = os.path.join(installDir, 'lib', 'site-packages')
+else:
+    versionedPython = 'python{0}.{1}'.format(
+        sys.version_info[0], sys.version_info[1])
+    packages = os.path.join(
+        installDir, 'lib', versionedPython, 'site-packages')
+
+
+class BinaryDistribution(Distribution):
+    def is_pure(self):
+        return False
+
+codaPyds = glob.glob(os.path.join(packages, 'coda', '_*pyd'))
+codaPyds = [os.path.basename(pyd) for pyd in codaPyds]
+
+sixPyds = glob.glob(os.path.join(packages, 'pysix', '_*pyd'))
+sixPyds = [os.path.basename(pyd) for pyd in sixPyds]
+
+
+setup(name = 'pysix',
+      version = '2.2.1',
```

Comment:
This is ready for review again.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Friday, September 02, 2016 9:15 AM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Add setup.py for creating wheels (#110)


In setup.py<https://github.com/ngageoint/six-library/pull/110#discussion_r77343636>:

> +        installDir, 'lib', versionedPython, 'site-packages')

> +

> +

> +class BinaryDistribution(Distribution):

> +    def is_pure(self):

> +        return False

> +

> +codaPyds = glob.glob(os.path.join(packages, 'coda', '_*pyd'))

> +codaPyds = [os.path.basename(pyd) for pyd in codaPyds]

> +

> +sixPyds = glob.glob(os.path.join(packages, 'pysix', '_*pyd'))

> +sixPyds = [os.path.basename(pyd) for pyd in sixPyds]

> +

> +

> +setup(name = 'pysix',

> +      version = '2.2.1',

From a synchronization perspective, I'd really like to have the version set in just one spot if possible so we don't forget to increment it (it used to be spread across multiple wscripts and was easy to get out of sync). Logically you'd want it in the top-level SIX wscript... but the problem is that the way we subtree pull this into other repos, and the way waf works, we won't actually call the top-level wscript when we're in that other repo. So for that reason, it's actually set in six/six/modules/wscript right now.

Any thoughts on how we can keep this to just one version? If it's completely impossible / really problematic, we can do it as two...

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ngageoint/six-library/pull/110/files/7ae935a30d887fb8b8f7eae0ae7213bbd7699c47#r77343636>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYMIXzauwHTs_aBkurHtxl1EFS-X2ks5qmCFSgaJpZM4JzsKy>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.